### PR TITLE
Convert AugmentR to use infra/core 

### PR DESCRIPTION
**Pull request by dotnet-7**
Convert AugmentR to use infra/core.
Now that we need to output the openai key for azd environment to use , we reserve the previous method to get it. 

@jongio for notification.

### CODE REVIEW

**Comment by jongio** [comment]
Can you please introduce a KV to this sample and store/pull the key from there?

**Comment by v-jiaodi** [comment]
@jongio KV has been introduced to this sample, please review.

**Comment by dotnet-7** [comment]
@jongio  - Updated according to your comments, please re-review this pr. Thanks!

File: Backend/Backend.csproj [hunk 1]
```diff
@@ -9,6 +9,8 @@
 	</PropertyGroup>
```
Comment (jongio) [inline Backend/Backend.csproj:9]: Please review the steps in the readme.  I see it does a clone, we should be using `azd init -t ` instead.

Comment (jongio) [reply Backend/Backend.csproj:9]: Also include an install link to azd in the readme.

**Review by bradygaster — APPROVED** [review]

**Pull request merged**